### PR TITLE
JHipster 7 upgrade

### PR DESCRIPTION
**Pull request by avdev4j**
close https://github.com/jhipster/generator-jhipster-blueprint/issues/200

### CODE REVIEW

File: .github/workflows/github-ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
             - name: 'Install dependencies'
               run: |
                   set -e
-                  npm install -g yo generator-jhipster
+                  npm install -g yo generator-jhipster@7.0.0-beta.1
```
Comment (mshima) [inline .github/workflows/github-ci.yml:34]:
```suggestion
                  npm install -g yo generator-jhipster@beta
```

File: generators/app/index.js [hunk 2]
```diff
@@ -54,7 +54,8 @@ module.exports = class extends Generator {
             .package()
             .then(
                 pkg => {
-                    this.jhipsterVersion = pkg.version;
+                    // this.jhipsterVersion = pkg.version;
+                    this.jhipsterVersion = '7.0.0-beta.1';
```
Comment (mshima) [inline generators/app/index.js:58]: Change to `repo('generator-jhipster@beta')` instead.

File: generators/app/templates/_generator.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Generator
+on:
+  push:
+    branches-ignore:
+      - 'dependabot/**'
```
Comment (mshima) [inline generators/app/templates/_generator.yml:5]: Why ignoring dependabot?
```suggestion
  push:
```

File: generators/app/templates/_generator.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    name: npm-test
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2.1.2
+        with:
+          node-version: '12'
+      - run: npm ci
+      - run: npm link
```
Comment (mshima) [inline generators/app/templates/_generator.yml:20]:
```suggestion
```

File: generators/app/templates/_generator.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    branches:
+      - '*'
+jobs:
+  generator-jhipster-<%= moduleName %>:
+    name: npm-test
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2.1.2
```
Comment (mshima) [inline generators/app/templates/_generator.yml:16]: v2.1.2 is beta, in this case we should use v2 tag instead.
```suggestion
      - uses: actions/setup-node@v2
```

**Review by mshima — COMMENTED** [review]
Would be nice to convert to async functions.

File: generators/app/templates/_generator.yml (hunk 5 above)
Comment (avdev4j) [reply generators/app/templates/_generator.yml]: great! thanks

File: generators/app/templates/_generator.yml (hunk 3 above)
Comment (avdev4j) [reply generators/app/templates/_generator.yml]: Good question.

**Comment by avdev4j** [comment]
@ruddell I have added all the sub-gen in this PR. I don't think we will do a release before the v7 release.

**Comment by avdev4j** [comment]
@mshima @murdos 
The Heroku test goes in timeout here when it works on my environment, any idea?

**Comment by mshima** [comment]
@avdev4j github actions contains heroku installed, so it's actually executing it.
No clue what it executes.
Vue page generator currently requires an exiting project, I will try to fix it.

**Comment by avdev4j** [comment]
> @avdev4j github actions contains heroku installed, so it's actually executing it.
Lol was not aware of that but it makes sense now to understand why it was so long to be executed.

**Comment by pascalgrimaud** [comment]
can you solve the conflicts plz @avdev4j ?
so we can merge it and prepare the release, just after v7 release

**Comment by avdev4j** [comment]
Done! @pascalgrimaud :)

**Pull request merged**